src/components/ReanimatedDrawerLayout.tsx

```diff
@@ -0,0 +1,745 @@
+      pointerEvents: isDrawerShowing.value
+        ? ('auto' as const)
+        : ('none' as const),
+    }));
+
+    // While the drawer is hidden, it's hitSlop overflows onto the main view by edgeWidth
+    // This way it can be swiped open even when it's hidden
+    const [edgeHitSlop, setEdgeHitSlop] = React.useState<HitSlop>(
+      isFromLeft ? { right: edgeWidth } : { left: edgeWidth }
+    );
+
+    // gestureOrientation is 1 if the gesture is expected to move from left to right and -1 otherwise
+    const gestureOrientation = React.useMemo(
+      () => (isFromLeft ? 1 : -1) * (drawerOpened ? -1 : 1),
+      [isFromLeft, drawerOpened]
+    );
+
+    const animateDrawer = React.useCallback(
+      (toValue: number, velocity: number, speed?: number) => {
+        'worklet';
+        const willShow = toValue !== 0;
+        isDrawerOpen.value = willShow;
+        runOnJS(setEdgeHitSlop)(
+          isFromLeft
+            ? { left: 0, width: isDrawerOpen.value ? undefined : edgeWidth }
+            : { right: 0, width: isDrawerOpen.value ? undefined : edgeWidth }
+        );
+
+        isDrawerShowing.value = willShow;
+
+        emitStateChanged(SETTLING, willShow);
+        runOnJS(setDrawerState)(SETTLING);
+
+        if (props.hideStatusBar) {
+          StatusBar.setHidden(willShow, props.statusBarAnimation || 'slide');
+        }
+
+        drawerTranslation.value = withSpring(
+          toValue,
+          {
+            // Velocity threshold does not matter as long as the destination is reached
+            // This prevents rubberbanding
+            restDisplacementThreshold: 1,
+            restSpeedThreshold: 10000,
+            overshootClamping: true,
+
+            velocity,
+            mass: speed ? 1 / speed : 2,
+            damping: 80,
+            stiffness: 500,
+          },
+          (finished) => {
+            if (finished) {
+              emitStateChanged(IDLE, willShow);
+              runOnJS(setDrawerOpened)(willShow);
+              runOnJS(setDrawerState)(IDLE);
+              if (willShow) {
+                dragX.value = drawerWidth * sideCorrection;
+                props.onDrawerOpen && runOnJS(props.onDrawerOpen)?.();
+              } else {
+                dragX.value = 0;
+                props.onDrawerClose && runOnJS(props.onDrawerClose)?.();
+              }
+            }
+          }
+        );
+      },
+      [
+        dragX,
+        drawerTranslation,
+        edgeWidth,
+        emitStateChanged,
+        isFromLeft,
+        isDrawerOpen,
+        props.hideStatusBar,
+        props.onDrawerClose,
+        props.onDrawerOpen,
+        props.statusBarAnimation,
+        drawerWidth,
+        sideCorrection,
+        isDrawerShowing,
+      ]
+    );
+
+    const handleRelease = React.useCallback(
+      ({
+        nativeEvent,
+      }: {
+        nativeEvent: {
+          translationX: number;
+          velocityX: number;
+          x: number;
+        };
```

Comment:
Why pack data into `nativeEvent` when it can be passed directly?

Reply:
Fixed in [5830645](https://github.com/software-mansion/react-native-gesture-handler/pull/3146/commits/58306453e30ac6e5d03bdfdc0b3be9d6526ef131)